### DISCUSSION TITLE
Microsoft.SqlServer.Rmo.dll .NET Library and DBATools Replication Feature

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [dataplat/dbatools#9281] [General]
Dear forum members,

I have been working with the DBATools modules and noticed that its replication feature is built upon the Microsoft.SqlServer.Rmo.dll .NET library. While exploring the documentation for Microsoft.SqlServer.Rmo.dll, I encountered some concerns that I would like to discuss with the community.

Firstly, when I try to browse the Microsoft.SqlServer.Rmo.dll .NET documentation and select SQL Server 2019 from the version dropdown menu, it automatically redirects me to the SQL Server 2016 version. This raises the question: Has the Microsoft.SqlServer.Rmo.dll library not been actively developed or updated in recent years to support the latest SQL Server versions?

If this is the case, it leads to another concern. Does this mean that the functions we have created in DBATools may not be able to fully leverage the features and capabilities of the recent versions of SQL Server? I have already noticed some discrepancies between the Microsoft.SqlServer.Rmo.dll documentation and SQL Server's documentation.

For example, in the Microsoft.SqlServer.Rmo.dll documentation page about the Articles schema option, there are 45 options listed. However, when I refer to SQL Server's official documentation about the article schema options, there are two additional options mentioned (E.g, 0x800000000000, Replicates any flitered nonclustered columnstore indexes on the tables).

I look forward to hearing your thoughts, experiences, and any potential solutions or recommendations.

Thank you.

**Comment by wsmelton** [comment]
> This raises the question: Has the Microsoft.SqlServer.Rmo.dll library not been actively developed or updated in recent years to support the latest SQL Server versions?

SQL Server teams answer would be something like: if it ain't broke don't change it 😉 

Officially I expect an answer of there have been no changes in SQL Server since 2016 that has required code changes, which in turn don't necessitate generating new documentation. The documentation framework used to render .NET/dotnet docs I don't think uses the "applies to" header like Azure and other Microsoft product docs do...which would make it more clear.

**Reply by potatoqualitee** [reply]
Unfortunately, it's not an open process :/

https://github.com/microsoft/sqlmanagementobjects/issues/59
